### PR TITLE
feat: newgrounds song loader via IDs

### PR DESCRIPTION
**Pull request by niceEli**

### CODE REVIEW

**Comment by niceEli** [comment]
id be good if there was some sort of config for the final download domain but idk how to do that

**Comment by amyspark-ng** [comment]
i approve 👍

**Comment by niceEli** [comment]
i had broken the jsdoc in types.ts so i fixed in [83a99c6](https://github.com/lajbel/newgrounds-js/pull/3/commits/83a99c6d2ad4ff34fe592cb5447ca55f6a221f64)

**Comment by niceEli** [comment]
> id be good if there was some sort of config for the final download domain but idk how to do that

i forgot to write but ive also solved this with an option in connect()

File: src/helpers/songs.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import { getClient } from "../helpers";
+/*
+ **
+ **  Created By: niceEli & SPCFork
+ **  License: MIT
+ **
+ **  Used In KaboomTS Before NGJS
+ **
+ */
```
Comment (lajbel) [inline src/helpers/songs.ts]: Is the comment necessary? I can add you in the README if you want

File: src/helpers/util.ts
Comment (lajbel) [inline src/helpers/util.ts:1]: Not necessary format changes

File: src/helpers.ts
Comment (lajbel) [inline src/helpers.ts:1]: I'm not a fan of this function shortening, please avoid change these stuff

File: src/client.ts [hunk 2]
```diff
@@ -4,30 +4,31 @@ import type { NGIOResponse } from "./types/objectModels";
...
-            const startSession = async () => {
+        !this._sessionID &&
+            (async function startSession() {
                 const sessionStarted = await this.call("App.startSession");
                 this._sessionID = sessionStarted.result.data.session.id;
-            };
-
-            startSession();
-        }
+            })();
```
Comment (lajbel) [inline src/client.ts]: The same about function shortening

**Review by lajbel — CHANGES_REQUESTED** [review]

File: src/helpers/songs.ts (hunk 1 above)
Comment (niceEli) [reply src/helpers/songs.ts]: fixing rn

Comment (niceEli) [reply src/helpers/songs.ts]: fixed in 8e802578ff1e81b20a9dd528478a070298c4e490

File: src/helpers/util.ts
Comment (niceEli) [reply src/helpers/util.ts:1]: fixed in [8f7666f](https://github.com/lajbel/newgrounds-js/pull/3/commits/8f7666f7d576984ac3c82a7e0b50a1d7102c9a24)

File: src/helpers.ts
Comment (niceEli) [reply src/helpers.ts:1]: fixed in [8f7666f](https://github.com/lajbel/newgrounds-js/pull/3/commits/8f7666f7d576984ac3c82a7e0b50a1d7102c9a24)

File: src/client.ts (hunk 2 above)
Comment (niceEli) [reply src/client.ts]: fixed in [8f7666f](https://github.com/lajbel/newgrounds-js/pull/3/commits/8f7666f7d576984ac3c82a7e0b50a1d7102c9a24)

**Review by lajbel — APPROVED** [review]
Looking good!

**Pull request merged**